programs/drift/src/math/liquidation.rs

```diff
@@ -323,6 +323,38 @@ pub fn validate_transfer_satisfies_limit_price(
     )
 }
 
-pub fn calculate_one_quote_worth_of_token(oracle_price: i64, decimals: u32) -> DriftResult<u128> {
-    10_u128.pow(decimals + 6).safe_div(oracle_price.cast()?)
+pub fn calculate_margin_shortage(
+    margin_requirement_with_buffer: u128,
+    total_collateral: i128,
+) -> DriftResult<u128> {
+    Ok(margin_requirement_with_buffer
+        .cast::<i128>()?
+        .safe_sub(total_collateral)?
+        .unsigned_abs())
+}
+
+pub fn calculate_max_pct_to_liquidate(
+    user: &User,
+    margin_shortage: u128,
+    slot: u64,
+    initial_pct_allowed_to_liquidate: u128,
```

Comment:
i see `initial_pct_allowed_to_liquidate` and `initial_pct_to_liquidate` interchangeable, probably best to pick one if its unchanged so its easier to follow